tmt/result.py

```diff
@@ -347,6 +368,21 @@ def interpret_result(self, interpret: ResultInterpret) -> 'Result':
             raise tmt.utils.SpecificationError(
                 f"Test result note '{self.note}' must be a string.")
 
+        # Interpret check results
+        failed_checks: list[CheckResult] = []
+        for check in self.check:
+            interpreted_check = check.interpret_check_result()
+            if (
+                    interpreted_check.result == ResultOutcome.FAIL and
+                    interpreted_check.check_result != CheckResultInterpret.INFO
+                    ):
+                failed_checks.append(interpreted_check)
```

Comment:
This will need a bit more work, because `self.check` is left untouched: interpretation happens, but `interpreted_check` is then thrown away, it's not stored anywhere. It's basically swapping one list for another, simple list comprehension. Together with the necessity of accessing checks to get their `result` key, it might be something like the following:

```python
# This is constructed in `from_test_invocation()` and fed to `interpret_result()` - just like the result interpretation:
{
    check.how: check.result
    for check in invocation.test.check
}


def interpret_result(self, interpret: ResultInterpret, interpret_checks: dict[str, CheckResultInterpret]) -> 'Result':
...

self.check = [
    check_result.interpret_check_result(interpret_checks[check_result.name])
    for check_result in self.check
]

failed_checks = [
    check_result
    for check_result in self.check
    if check_result.outcome == ResultOutcome.FAIL]
```

Reply:
Thanks for the review. Needs work, yes. Though I might need someone to go through the code with me, because most of the time I'm just guessing what is supposed to do what and how and then trying a blind trial-error. Not enjoyable and I'm back to feeling like an useless idiot.

Reply:
Well, I did offer to have an online session, and the offer still stands. Add something to my calendar and we can try it out.

Reply:
awesome, much appreciated.